### PR TITLE
Perform rename tests for all non nested folders 

### PR DESCRIPTION
**Pull request by anushka567**
### Description
The [_perform_testing](https://github.com/GoogleCloudPlatform/gcsfuse/blob/0b9cc168538bffd2ce6575ddffa32cb0a0b560e7/perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py#L103) function mounts the gcs bucket based on the test type(flat or hns) and then makes calls to record time taken to rename all the folders present in the "folders" section of dir JSON object.
The [_record_time_of_operation](https://github.com/GoogleCloudPlatform/gcsfuse/blob/0b9cc168538bffd2ce6575ddffa32cb0a0b560e7/perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py#L81) makes call to the [_record_time_for_folder_rename](https://github.com/GoogleCloudPlatform/gcsfuse/blob/0b9cc168538bffd2ce6575ddffa32cb0a0b560e7/perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py#L41) function  for each folder to record the time taken for rename operations performed on all the folders passed via directory JSON object
### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - Unit test for
a. [_record_time_of_operation](https://github.com/GoogleCloudPlatform/gcsfuse/blob/0b9cc168538bffd2ce6575ddffa32cb0a0b560e7/perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark_test.py#L43)
b. [_perform_testing](https://github.com/GoogleCloudPlatform/gcsfuse/blob/0b9cc168538bffd2ce6575ddffa32cb0a0b560e7/perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark_test.py#L81)

4. Integration tests - NA

### CODE REVIEW

File: perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py [hunk 1]
```diff
@@ -97,6 +97,50 @@ def _record_time_for_folder_rename(mount_point,folder,num_samples):
...
+    # Mounting the flat bucket.
+    flat_mount_flags = "--implicit-dirs --rename-dir-limit=1000000"
+    flat_bucket = mount_gcs_bucket(dir["name"], flat_mount_flags, log)
+
+    # Record time of operation and populate the results dict.
+    flat_results = _record_time_of_operation(flat_bucket, dir, num_samples)
+    results["flat"] = flat_results
+
+    unmount_gcs_bucket(dir["name"], log)
+  elif test_type == "hns":
```
Comment (ankitaluthra1) [inline perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py:136]: Default can be flat, In the start of method we can add if check for hns, there is no need for else, return from inside if hns.

Comment (anushka567) [reply perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py]: Updated.

**Review by ankitaluthra1 — DISMISSED** [review]

File: perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py [hunk 2]
```diff
@@ -78,6 +78,47 @@ def _record_time_for_folder_rename(mount_point,folder,num_samples):
...
+  """
+  This function performs rename operations and records time of operation .
+  """
+  if test_type == "hns":
+    # TODO add mount function for test type hns
+    return
+
+  # Mounting the gcs bucket.
+  flat_mount_flags = "--implicit-dirs --rename-dir-limit=1000000"
+  flat_bucket = mount_gcs_bucket(dir["name"], flat_mount_flags, log)
```
Comment (Tulsishah) [inline perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py:113]: nit: flat_bucket_name

File: perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py [hunk 3]
```diff
@@ -78,6 +78,47 @@ def _record_time_for_folder_rename(mount_point,folder,num_samples):
...
+  # Collecting metrics for non-nested folders.
+  for folder in dir["folders"]["folder_structure"]:
+    results[folder["name"]] = _record_time_for_folder_rename(mount_point,folder,num_samples)
+  #TODO Add metric collection logic for nested-folders
+  return results
+
+
+def _perform_testing(dir, test_type, num_samples, results):
+  """
+  This function performs rename operations and records time of operation .
```
Comment (Tulsishah) [inline perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py:105]: Add argument format

File: perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark_test.py [hunk 4]
```diff
@@ -38,6 +38,79 @@ def test_record_time_for_folder_rename(self,mock_time,mock_subprocess):
...
+                {
+                    'name': "test_folder2",
+                    "num_files": 1,
+                    "file_name_prefix": "file",
+                    "file_size": "1kb"
+                }
+            ]
+        }
+    }
+
```
Comment (Tulsishah) [inline perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark_test.py:63]: nit: remove line

**Review by ashmeenkaur — APPROVED** [review]

**Pull request merged**